joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala

```diff
@@ -1,130 +1,755 @@
+      }
+    }
+
+    "create a typeDecl node inheriting from correct interface" in {
+      cpg.typeDecl.name(".*lambda.*").l match {
+        case List(lambdaDecl) =>
+          lambdaDecl.name shouldBe "lambda$0"
+          lambdaDecl.fullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Integer)"
+          lambdaDecl.inheritsFromTypeFullName should contain theSameElementsAs List("Foo")
+
+        case result => fail(s"Expected a single typeDecl for the lambda but got $result")
+      }
+    }
+
+    // TODO: Fix typeDecl for interfaceBinding
+    "create bindings to implemented method" in {
+      cpg.all.collectAll[Binding].nameExact("baz").sortBy(_.methodFullName).toList match {
+        case List(interfaceBinding, binding, erasedBinding) =>
+          interfaceBinding.methodFullName shouldBe "Foo.baz:java.lang.String(java.lang.Object,java.lang.Object)"
+          interfaceBinding.signature shouldBe "java.lang.String(java.lang.Object,java.lang.Object)"
+          interfaceBinding.inE.collectAll[Binds].map(_.outNode()).l match {
+            case List(typeDecl: TypeDecl) => typeDecl.fullName shouldBe "Foo"
+            case result                   => fail(s"Expected typeDecl but got $result")
+          }
+
+          binding.methodFullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Integer)"
+          binding.signature shouldBe "java.lang.String(java.lang.Integer,java.lang.Integer)"
+          binding.inE.collectAll[Binds].map(_.outNode()).l match {
+            case List(typeDecl: TypeDecl) =>
+              typeDecl.fullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Integer)"
+            case result => fail(s"Expected typeDecl but got $result")
+          }
+
+          erasedBinding.methodFullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Integer)"
+          erasedBinding.signature shouldBe "java.lang.String(java.lang.Object,java.lang.Object)"
+          erasedBinding.inE.collectAll[Binds].map(_.outNode()).l match {
+            case List(typeDecl: TypeDecl) =>
+              typeDecl.fullName shouldBe "TestClass.lambda$0:java.lang.String(java.lang.Integer,java.lang.Integer)"
+            case result => fail(s"Expected typeDecl but got $result")
+          }
+
+        case result => fail(s"Expected three bindings to baz method but got $result")
+      }
     }
   }
 
-  "CPG for `identity-like` supplier" - {
-    lazy val cpg = JavaSrc2CpgTestContext.buildCpg("""
+  // This is an example of a case where all the type info we need to figure out that the lambda implements
+  // `Function<String, String>`, but from JavaParser and the current expectedType propagation we only get
+  // `java.util.Function<Object, Object>`. Ideally we'd want all occurrences of `java.lang.Object` to be
+  // `java.lang.String` instead.
+  "a lambda implementing a mapper in stream" should {
```

Comment:
Is this another bug in JavaParser?

Reply:
I haven't looked too deeply into this, but as far as I can tell it's a limitation of the JavaParser type solver rather than a bug.